fw-child/resources/functions/rest.php

```diff
@@ -664,192 +664,211 @@ function cdc_get_idf_url () {
+
+            if ( $wp_mail ) {
+                $result['mail'] = 'success';
+                $result['header'] = __( 'Thanks! We’ve received your inquiry.', 'cdc' );
+                $result['messages'] = [ __( 'Please note: we are currently experiencing a higher than normal number of inquiries to our Support Desk. We will do our best to reply to you as soon as possible, but please be advised that there may be delays.', 'cdc' ) ];
+            } else {
+                $result['messages'][] = __( 'Something went wrong while sending your message. Please try again later.', 'cdc' );
+            }
+
+            // Mailchimp signup for feedback form
+            if ( $include_mailchimp && isset ( $form_data['signup'] ) && $form_data['signup'] == 'true' ) {
+                include_once ( locate_template ( 'resources/php/mailchimp.php' ) );
+                $signup = mailchimp_register ( $form_data['email'] );
+                if ( $signup == true ) {
+                    $result['signup'] = 'success';
+                    $result['messages'][] = __( 'Your email address has been added to our mailing list. You can unsubscribe at any time.', 'cdc' );
+                } elseif ( $signup == false ) {
+                    $result['signup'] = 'failed';
+                    $result['messages'][] = __( 'An error occurred while adding your email address to our mailing list.', 'cdc' );
+                }
+            }
+        }
+    } else {
+        // No form data
+        $valid = false;
+        $result['invalid'][] = 'form data';
+        $result['messages'][] = 'An error occurred.';
+    }
+
+    echo json_encode ( $result );
+}
+
+function build_message_body ( $form_data, $form_type ) {
+    $body = '';
+    $body .= '<p><span style="display: inline-block; width: 150px; font-weight: bold; vertical-align: top;">' . __( 'Name', 'cdc-feedback' ) . '</span><span style="display: inline-block; vertical-align: top;">' . $form_data['fullname'] . '</span></p>';
+    $body .= '<p><span style="display: inline-block; width: 150px; font-weight: bold; vertical-align: top;">' . __( 'Email', 'cdc-feedback' ) . '</span><span style="display: inline-block; vertical-align: top;"><a href="mailto:' . $form_data['email'] . '">' . $form_data['email'] . '</a></span></p>';
+    $body .= '<p><span style="display: inline-block; width: 150px; font-weight: bold; vertical-align: top;">' . __( 'Organization', 'cdc-feedback' ) . '</span><span style="display: inline-block; vertical-align: top;">' . $form_data['organization'] . '</span></p>';
+
+    // Custom fields for feedback form
+    if ( $form_type == 'feedback' ) {
+        $body .= build_feedback_body ( $form_data );
+    } 
+    // Custom fields for betaapps form
+    elseif ( $form_type == 'betaapps' ) {
+        $body .= build_betaapps_body ( $form_data );
+    }
+    
+    $body .= '<p style="font-weight: bold;">Message</p><p><pre>' . $form_data['feedback'] . '</pre></p>';
```

Comment:
Even though the browser will still render it, `<pre>` tags cannot be inside `<p>` tags, they are both block-level elements.

Reply:
Right.